homeassistant/config_entries.py

```diff
@@ -1528,6 +1531,99 @@ class OptionsFlow(data_entry_flow.FlowHandler):
     handler: str
 
 
+class HelperCommonFlowHandler:
+    """Handle a config or options flow for helper."""
+
+    def __init__(
+        self,
+        handler: HelperConfigFlowHandler | HelperOptionsFlowHandler,
+        config_entry: ConfigEntry | None,
+    ) -> None:
+        """Initialize a common handler."""
+        self._handler = handler
+        self._options = dict(config_entry.data) if config_entry is not None else {}
+
+    @staticmethod
+    async def validate_input(
+        hass: HomeAssistant, data: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Validate the user input allows us to connect."""
+
+        # TODO: Some fancy validation of the selectors?
+        return {"title": data["name"]}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the initial step."""
+        errors = None
+        if user_input is not None:
+            errors = {}
+            try:
+                info = await self.validate_input(self._handler.hass, user_input)
+            except Exception:
+                # TODO: Error handling
+                errors["base"] = "blabla"
+            else:
+                return self._handler.async_create_entry(
+                    title=info["title"], data=user_input
+                )
+
+        schema = copy.deepcopy(self._handler.schema.schema)
+        for key in schema:
+            if key in self._options:
+                key.description = {"suggested_value": self._options[key]}
```

Comment:
We probably want to check the key is a `vol.Marker` before setting a description on it